### ISSUE TITLE
🌸 Cherry-pick request for #36596 into 2110212130001

### ISSUE THREAD

**Issue by alanorozco** [ampproject/amphtml#36598]
### Issue (P0 Bug)

#36596

### Pull Request(s)

#36596

### Release Tracker(s)

?

### Channels

- Beta / Experimental (before we promote to stable)
- We should also cut a new Opt-in % Beta for this week instead of using a cut from last Friday.

### Formats

Websites

### Justification

Bento launch is blocked by breakage of `customElements.whenDefined`

### Verification Steps

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <!-- Fails without this script, which is created by PR: -->
    <!--
      <script
        src="https://cdn.ampproject.org/custom-elements-polyfill.js"
      ></script>
    -->
    <script async src="https://cdn.ampproject.org/bento.js"></script>
    <script
      async
      src="https://cdn.ampproject.org/v0/bento-fit-text-1.0.js"
    ></script>
  </head>
  <body>
    <bento-fit-text style="width: 80px; height: 30px">
      Lorem ipsum dolor sit amet
    </bento-fit-text>
    <script>
      (async () => {
        // Never resolves without custom-elements-polyfill.js:
        await customElements.whenDefined('bento-fit-text');
        console.log(document.querySelector('bento-fit-text'));
      })();
    </script>
  </body>
</html>
```

### Impact

No live AMP documents were affected.

### Action Items

- Add e2e test for `customElements.whenDefined`
- Verify that this test is run in both `module` and `nomodule` browsers

### Notifications

/cc @ampproject/release-on-duty @ampproject/wg-approvers @ampproject/cherry-pick-approvers
